internal/collector/recon/utils.go

```diff
@@ -96,13 +96,20 @@ func splitOutputPerHost(output []byte, cmdArgs []string) (map[string][]byte, err
 		data = bytes.ReplaceAll(data, []byte(`True`), []byte(`true`))
 		data = bytes.ReplaceAll(data, []byte(`False`), []byte(`false`))
 		data = bytes.ReplaceAll(data, []byte(`None`), []byte(`"None"`))
-
+		data = []byte(stripEscapeChars(string(data)))
```

Comment:
What is the purpose of this? This looks like it could break existing users of this function.

Reply:
During testing I ran into an issue where, after a storage node was taken down for maintenance during a benchmark test, the target container and account values returned by swift-recon had null characters prepended which broke the metric returns.

I was of the opinion that stripping the null characters and exporting what metrics were available from that container was more desirable than allowing it to fail.

I can do a few things here:

1. Move this functionality into the sharding module, leaving it out of the central utils file, however I haven't seen an instance in which this would break any of the other metrics.
2. Handle and log the exception, without proceeding.
3. Allow it to fail, and make this an operational task, in understanding why this occurs at the object store level.

Reply:
Oh, I'm guessing by "null characters" you mean ANSI escape codes? (Null characters are something very different from what the regex matches.) Do you have a capture of this, so we can include that as a testcase?

Reply:
Yes, the particular case I'm referencing started returning the escape code \x00 which as far as I understand denotes a null character.

I have already included the captured output as one of the main test cases here:
https://github.com/sapcc/swift-health-exporter/pull/122/files#diff-e4f18ca2adb9975ff8ecd3e8bffe52d4fa914bbfcb30a78f90817988ea6cee54R213

Reply:
Thanks for the link. After reading a bit into the Swift source code, this is not an error case. This is just how Swift legitimately formats sharding container names: https://github.com/openstack/swift/blob/8089925e776d0dd139d4d4a6dc8b5988b258ede3/test/probe/test_sharder.py#L705

So I don't think any stripping out is necessary at all. These are not actual NUL bytes anyway, just escape sequences. In fact, it is probably more wrong than not to mess with the legitimate container names.

Reply:
Thanks for the information, yes I agree that stripping part of the swift naming is not the right approach.

In this case, I will handle this so as not to strip the characters, but not break the json unmarshalling.

Reply:
As go interprets \x as an invalid escape, I've just replaced it with `\\x` to try to preserve the name while preventing the unmarshalling from failing.